### PR TITLE
Refactoring wallet list menu modal

### PR DESCRIPTION
**Pull request by axenov-it**
#### PR Requirements

If you have made **any** visual changes to the GUI. Make sure you have:
- [ ] Tested on iOS Tablet
- [ ] Tested on small Android
- [ ] n/a

### CODE REVIEW

**Comment by axenov-it** [comment]
This functionality have: 

1) Refactoring code with react hooks 
2) Added types for useDispatch and useSelector from react-redux lib 
3) Implemented splittable option for wallet list modal menu with account.listSplittableWalletTypes of handler

File: src/components/modals/WalletListMenuModal.js [hunk 1]
```diff
@@ -52,99 +44,78 @@ const icons = {
...
-      return
-    }
+  for (const option of WALLET_LIST_MENU) {
+    const { currencyCodes, label, value } = option
 
-    // Main wallet options
-    for (const option of WALLET_LIST_MENU) {
-      const { currencyCodes, label, value } = option
-      if (currencyCodes != null && !currencyCodes.includes(currencyCode)) continue
+    if (isArray(currencyCodes) && !currencyCodes.includes(currencyCode)) continue
```
Comment (thehobbit85) [inline src/components/modals/WalletListMenuModal.js:79]: Remove `lodash`. Just use `Array.isArray()`

File: src/components/modals/WalletListMenuModal.js [hunk 2]
```diff
@@ -57,6 +66,13 @@ const getWalletOptions = async (params: { walletId: string, walletName?: string,
 
   const result = []
 
+  const splittable = await account.listSplittableWalletTypes(walletId)
+
+  if (splittable.length) {
```
Comment (thehobbit85) [inline src/components/modals/WalletListMenuModal.js:71]: This is not a boolean, so you should use exact comparison, something like: 
`splittable.length !== null && splittable.length > 0`

**Review by thehobbit85 — CHANGES_REQUESTED** [review]
See comments

**Review by thehobbit85 — APPROVED** [review]

**Pull request merged**